Submitted By: Daniel Baumann (daniel.baumann@panthera-systems.net)
Date: 2004-07-10
Description: Adds support for parallelcompiling.

diff -Naur LFS-5.1.1.original/LFS.xml LFS-5.1.1/LFS.xml
--- LFS-5.1.1.original/LFS.xml	2004-06-03 06:20:29.000000000 +0200
+++ LFS-5.1.1/LFS.xml	2004-07-10 11:15:32.000000000 +0200
@@ -42,6 +42,7 @@
 				<variable name="LFS">&LFS;</variable>
 				<variable name="LC_ALL">POSIX</variable>
 				<variable name="PATH" mode="prepend">/tools/bin:</variable>
+				<variable name="CC_PARALLEL">&gcc_parallel;</variable>
 				<variable name="CFLAGS">&gcc_optimizations;</variable>
 				<variable name="CXXFLAGS">&gcc_optimizations;</variable>
 				<variable name="CC"></variable>
@@ -96,6 +97,7 @@
 				<variable name="TERM">&TERM;</variable>
 				<variable name="PS1">'\u:\w\$ '</variable>
 				<variable name="PATH">/bin:/usr/bin:/sbin:/usr/sbin:/tools/bin</variable>
+				<variable name="CC_PARALLEL">&gcc_parallel;</variable>
 				<variable name="CFLAGS">&gcc_optimizations;</variable>
 				<variable name="CXXFLAGS">&gcc_optimizations;</variable>
 			</environment>
@@ -169,6 +171,7 @@
 			<variable name="TERM">&TERM;</variable>
 			<variable name="PS1">'\u:\w\$ '</variable>
 			<variable name="PATH">/bin:/usr/bin:/sbin:/usr/sbin</variable>
+			<variable name="CC_PARALLEL">&gcc_parallel</variable>
 			<variable name="CFLAGS">&gcc_optimizations;</variable>
 			<variable name="CXXFLAGS">&gcc_optimizations;</variable>
 		</environment>
@@ -189,6 +192,7 @@
 			<variable name="TERM">&TERM;</variable>
 			<variable name="PS1">'\u:\w\$ '</variable>
 			<variable name="PATH">/bin:/usr/bin:/sbin:/usr/sbin</variable>
+			<variable name="CC_PARALLEL">&gcc_paralle;</variable>
 			<variable name="CFLAGS">&gcc_optimizations;</variable>
 			<variable name="CXXFLAGS">&gcc_optimizations;</variable>
 		</environment>
@@ -207,6 +211,7 @@
                         <variable name="TERM">&TERM;</variable>
                         <variable name="PS1">'\u:\w\$ '</variable>
                         <variable name="PATH">/bin:/usr/bin:/sbin:/usr/sbin</variable>
+			<variable name="CC_PARALLEL">&gcc_parallel;</variable>
                         <variable name="CFLAGS">&gcc_optimizations;</variable>
                         <variable name="CXXFLAGS">&gcc_optimizations;</variable>
                 </environment>
diff -Naur LFS-5.1.1.original/chapter05/binutils-pass1.xml LFS-5.1.1/chapter05/binutils-pass1.xml
--- LFS-5.1.1.original/chapter05/binutils-pass1.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/binutils-pass1.xml	2004-07-10 11:15:32.000000000 +0200
@@ -15,6 +15,7 @@
 	<stage name="Installing.">
 		<stageinfo>
 			<environment>
+				<variable name="CC_PARALLEL" />
 				<variable name="CFLAGS" />
 				<variable name="CXXFLAGS" />
 			</environment>
@@ -31,6 +32,7 @@
 		</make>
 
 		<make>
+			<param>-j $CC_PARALLEL</param>
 			<param>LDFLAGS="-all-static"</param>
 		</make>
 
@@ -44,6 +46,7 @@
 		</make>
 
 		<make>
+			<param>-j $CC_PARALLEL</param>
 			<param>-C ld</param>
 			<param>LDFLAGS="-all-static"</param>
 			<param>LIB_PATH=/tools/lib</param>
diff -Naur LFS-5.1.1.original/chapter05/binutils-pass2.xml LFS-5.1.1/chapter05/binutils-pass2.xml
--- LFS-5.1.1.original/chapter05/binutils-pass2.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/binutils-pass2.xml	2004-07-10 11:15:32.000000000 +0200
@@ -15,6 +15,7 @@
 	<stage name="Installing.">
 		<stageinfo>
 			<environment>
+				<variable name="CC_PARALLEL" />
 				<variable name="CFLAGS" />
 				<variable name="CXXFLAGS" />
 			</environment>
@@ -27,7 +28,9 @@
 			<param>--with-lib-path=/tools/lib</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter05/bzip2.xml LFS-5.1.1/chapter05/bzip2.xml
--- LFS-5.1.1.original/chapter05/bzip2.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/bzip2.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,10 +9,14 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&bzip2-directory;</base>
 		</stageinfo>
 
 		<make>
+			<param>-j $CC_PARALLEL</param>
 			<param>PREFIX=/tools</param>
 			<param>install</param>
 		</make>
diff -Naur LFS-5.1.1.original/chapter05/coreutils.xml LFS-5.1.1/chapter05/coreutils.xml
--- LFS-5.1.1.original/chapter05/coreutils.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/coreutils.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&coreutils-directory;</base>
 		</stageinfo>
 
@@ -17,7 +20,9 @@
 			<param>--prefix=/tools</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter05/dejagnu.xml LFS-5.1.1/chapter05/dejagnu.xml
--- LFS-5.1.1.original/chapter05/dejagnu.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/dejagnu.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&dejagnu-directory;</base>
 		</stageinfo>
 
@@ -17,6 +20,7 @@
 		</configure>
 
 		<make>
+			<param> -j $CC_PARALLEL</param>
 			<param>install</param>
 		</make>
 	</stage>
diff -Naur LFS-5.1.1.original/chapter05/diffutils.xml LFS-5.1.1/chapter05/diffutils.xml
--- LFS-5.1.1.original/chapter05/diffutils.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/diffutils.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&diffutils-directory;</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/tools</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter05/expect.xml LFS-5.1.1/chapter05/expect.xml
--- LFS-5.1.1.original/chapter05/expect.xml	2004-05-18 06:59:04.000000000 +0200
+++ LFS-5.1.1/chapter05/expect.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&expect-directory;</base>
 		</stageinfo>
 
@@ -24,7 +27,9 @@
 			<param>--with-x=no</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter05/findutils.xml LFS-5.1.1/chapter05/findutils.xml
--- LFS-5.1.1.original/chapter05/findutils.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/findutils.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&findutils-directory;</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/tools</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter05/gawk.xml LFS-5.1.1/chapter05/gawk.xml
--- LFS-5.1.1.original/chapter05/gawk.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/gawk.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&gawk-directory;</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/tools</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter05/gcc-pass1.xml LFS-5.1.1/chapter05/gcc-pass1.xml
--- LFS-5.1.1.original/chapter05/gcc-pass1.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/gcc-pass1.xml	2004-07-10 11:15:32.000000000 +0200
@@ -15,6 +15,7 @@
 	<stage name="Installing.">
 		<stageinfo>
 			<environment>
+				<variable name="CC_PARALLEL" />
 				<variable name="CFLAGS" />
 				<variable name="CXXFLAGS" />
 			</environment>
@@ -30,6 +31,7 @@
 		</configure>
 
 		<make>
+			<param>-j $CC_PARALLEL</param>
 			<param>BOOT_LDFLAGS="-static"</param>
 			<param>bootstrap</param>
 		</make>
diff -Naur LFS-5.1.1.original/chapter05/gcc-pass2.xml LFS-5.1.1/chapter05/gcc-pass2.xml
--- LFS-5.1.1.original/chapter05/gcc-pass2.xml	2004-05-18 06:59:04.000000000 +0200
+++ LFS-5.1.1/chapter05/gcc-pass2.xml	2004-07-10 11:15:32.000000000 +0200
@@ -27,6 +27,7 @@
 	<stage name="Installing.">
 		<stageinfo>
 			<environment>
+				<variable name="CC_PARALLEL" />
 				<variable name="CFLAGS" />
 				<variable name="CXXFLAGS" />
 			</environment>
@@ -62,7 +63,9 @@
 			<param>--enable-languages=c,c++</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter05/gettext.xml LFS-5.1.1/chapter05/gettext.xml
--- LFS-5.1.1.original/chapter05/gettext.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/gettext.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&gettext-directory;</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/tools</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter05/glibc.xml LFS-5.1.1/chapter05/glibc.xml
--- LFS-5.1.1.original/chapter05/glibc.xml	2004-05-18 08:02:35.000000000 +0200
+++ LFS-5.1.1/chapter05/glibc.xml	2004-07-10 11:15:32.000000000 +0200
@@ -15,6 +15,7 @@
 	<stage name="Installing.">
 		<stageinfo>
 			<environment>
+				<variable name="CC_PARALLEL" />
 				<variable name="CFLAGS" />
 				<variable name="CXXFLAGS" />
 			</environment>
@@ -32,6 +33,7 @@
 		</configure>
 
 		<make>
+			<param>PARALLELMFLAGS="-j $CC_PARALLEL"</param>
 			<param>AUTOCONF=no</param>
 		</make>
 
diff -Naur LFS-5.1.1.original/chapter05/grep.xml LFS-5.1.1/chapter05/grep.xml
--- LFS-5.1.1.original/chapter05/grep.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/grep.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,8 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
 			<base>&build_dir;/&grep-directory;</base>
 		</stageinfo>
 
@@ -18,7 +20,9 @@
 			<param>--with-included-regex</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter05/gzip.xml LFS-5.1.1/chapter05/gzip.xml
--- LFS-5.1.1.original/chapter05/gzip.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/gzip.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</enviroment>
 			<base>&build_dir;/&gzip-directory;</base>
 		</stageinfo>
 
diff -Naur LFS-5.1.1.original/chapter05/lockingglibc.xml LFS-5.1.1/chapter05/lockingglibc.xml
--- LFS-5.1.1.original/chapter05/lockingglibc.xml	2004-05-04 00:52:59.000000000 +0200
+++ LFS-5.1.1/chapter05/lockingglibc.xml	2004-07-10 11:15:32.000000000 +0200
@@ -1,10 +1,13 @@
 <stage name="Locking glibc.">
 	<stage name="Adjusting the linker.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
 			<base>&build_dir;/binutils-build</base>
 		</stageinfo>
 
 		<make>
+			<param>-j $CC_PARALLEL</param>
 			<param>-C ld</param>
 			<param>install</param>
 		</make>
diff -Naur LFS-5.1.1.original/chapter05/make.xml LFS-5.1.1/chapter05/make.xml
--- LFS-5.1.1.original/chapter05/make.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/make.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</enviroment>
 			<base>&build_dir;/&make-directory;</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/tools</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter05/ncurses.xml LFS-5.1.1/chapter05/ncurses.xml
--- LFS-5.1.1.original/chapter05/ncurses.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/ncurses.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&ncurses-directory;</base>
 		</stageinfo>
 
@@ -20,7 +23,13 @@
 			<param>--enable-overwrite</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>sources</param>
+		</make>
+
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter05/patch.xml LFS-5.1.1/chapter05/patch.xml
--- LFS-5.1.1.original/chapter05/patch.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/patch.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&patch-directory;</base>
 		</stageinfo>
 
@@ -17,7 +20,9 @@
 			<param>--prefix=/tools</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter05/perl.xml LFS-5.1.1/chapter05/perl.xml
--- LFS-5.1.1.original/chapter05/perl.xml	2004-05-18 06:59:04.000000000 +0200
+++ LFS-5.1.1/chapter05/perl.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&perl-directory;</base>
 		</stageinfo>
 
@@ -35,6 +38,7 @@
 		</configure>
 
 		<make>
+			<param>-j $CC_PARALLEL</param>
 			<param>perl</param>
 			<param>utilities</param>
 		</make>
diff -Naur LFS-5.1.1.original/chapter05/sed.xml LFS-5.1.1/chapter05/sed.xml
--- LFS-5.1.1.original/chapter05/sed.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/sed.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&sed-directory;</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/tools</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter05/tar.xml LFS-5.1.1/chapter05/tar.xml
--- LFS-5.1.1.original/chapter05/tar.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/tar.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&tar-directory;</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/tools</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter05/tcl.xml LFS-5.1.1/chapter05/tcl.xml
--- LFS-5.1.1.original/chapter05/tcl.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/tcl.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&tcl-directory;/unix</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/tools</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter05/texinfo.xml LFS-5.1.1/chapter05/texinfo.xml
--- LFS-5.1.1.original/chapter05/texinfo.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/texinfo.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&texinfo-directory;</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/tools</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter05/util-linux.xml LFS-5.1.1/chapter05/util-linux.xml
--- LFS-5.1.1.original/chapter05/util-linux.xml	2004-05-16 01:46:47.000000000 +0200
+++ LFS-5.1.1/chapter05/util-linux.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&util-linux-directory;</base>
 		</stageinfo>
 
@@ -21,16 +24,19 @@
 		<configure />
 
 		<make>
+			<param>-j $CC_PARALLEL</param>
 			<param>-C lib</param>
 		</make>
 
 		<make>
+			<param>-j $CC_PARALLEL</param>
 			<param>-C mount</param>
 			<param>mount</param>
 			<param>umount</param>
 		</make>
 
 		<make>
+			<param>-j $CC_PARALLEL</param>
 			<param>-C text-utils</param>
 			<param>more</param>
 		</make>
diff -Naur LFS-5.1.1.original/chapter06/adjustingtoolchain.xml LFS-5.1.1/chapter06/adjustingtoolchain.xml
--- LFS-5.1.1.original/chapter06/adjustingtoolchain.xml	2004-05-04 01:02:22.000000000 +0200
+++ LFS-5.1.1/chapter06/adjustingtoolchain.xml	2004-07-10 11:15:32.000000000 +0200
@@ -1,10 +1,14 @@
 <stage name="Re-adjusting the toolchain.">
 	<stage name="Adjusting the linker.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/binutils-build</base>
 		</stageinfo>
 
 		<make>
+			<param>-j $CC_PARALLEL</param>
 			<param>-C ld</param>
 			<param>INSTALL=/tools/bin/install</param>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter06/autoconf.xml LFS-5.1.1/chapter06/autoconf.xml
--- LFS-5.1.1.original/chapter06/autoconf.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/autoconf.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&autoconf-directory;</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/usr</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter06/automake.xml LFS-5.1.1/chapter06/automake.xml
--- LFS-5.1.1.original/chapter06/automake.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/automake.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&automake-directory;</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/usr</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter06/binutils.xml LFS-5.1.1/chapter06/binutils.xml
--- LFS-5.1.1.original/chapter06/binutils.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/binutils.xml	2004-07-10 11:15:32.000000000 +0200
@@ -15,6 +15,7 @@
 	<stage name="Installing.">
 		<stageinfo>
 			<environment>
+				<variable name="CC_PARALLEL" />
 				<variable name="CFLAGS" />
 				<variable name="CXXFLAGS" />
 			</environment>
@@ -34,6 +35,7 @@
 		</configure>
 
 		<make>
+			<param>-j $CC_PARALLEL</param>
 			<param>tooldir=/usr</param>
 		</make>
 
diff -Naur LFS-5.1.1.original/chapter06/bison.xml LFS-5.1.1/chapter06/bison.xml
--- LFS-5.1.1.original/chapter06/bison.xml	2004-05-18 06:59:04.000000000 +0200
+++ LFS-5.1.1/chapter06/bison.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&bison-directory;</base>
 		</stageinfo>
 
@@ -22,7 +25,9 @@
 			<param>--prefix=/usr</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter06/bzip2.xml LFS-5.1.1/chapter06/bzip2.xml
--- LFS-5.1.1.original/chapter06/bzip2.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/bzip2.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&bzip2-directory;</base>
 		</stageinfo>
 
@@ -20,7 +23,9 @@
 			<param>clean</param>
 		</make>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter06/coreutils.xml LFS-5.1.1/chapter06/coreutils.xml
--- LFS-5.1.1.original/chapter06/coreutils.xml	2004-05-18 06:59:04.000000000 +0200
+++ LFS-5.1.1/chapter06/coreutils.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&coreutils-directory;</base>
 		</stageinfo>
 
@@ -29,7 +32,9 @@
 			<param>--prefix=/usr</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 <!--
 		<make>
 			<param>install-root</param>
diff -Naur LFS-5.1.1.original/chapter06/diffutils.xml LFS-5.1.1/chapter06/diffutils.xml
--- LFS-5.1.1.original/chapter06/diffutils.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/diffutils.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&diffutils-directory;</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/usr</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter06/e2fsprogs.xml LFS-5.1.1/chapter06/e2fsprogs.xml
--- LFS-5.1.1.original/chapter06/e2fsprogs.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/e2fsprogs.xml	2004-07-10 11:15:32.000000000 +0200
@@ -13,6 +13,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/e2fsprogs-build</base>
 		</stageinfo>
 
@@ -22,7 +25,9 @@
 			<param>--enable-elf-shlibs</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter06/ed.xml LFS-5.1.1/chapter06/ed.xml
--- LFS-5.1.1.original/chapter06/ed.xml	2004-05-18 06:59:04.000000000 +0200
+++ LFS-5.1.1/chapter06/ed.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+ 			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&ed-directory;</base>
 		</stageinfo>
 
@@ -23,7 +26,9 @@
 			<param>--exec-prefix=""</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter06/file.xml LFS-5.1.1/chapter06/file.xml
--- LFS-5.1.1.original/chapter06/file.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/file.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&file-directory;</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/usr</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter06/findutils.xml LFS-5.1.1/chapter06/findutils.xml
--- LFS-5.1.1.original/chapter06/findutils.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/findutils.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&findutils-directory;</base>
 		</stageinfo>
 
@@ -18,7 +21,9 @@
 			<param>--localstatedir=/var/lib/misc</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter06/flex.xml LFS-5.1.1/chapter06/flex.xml
--- LFS-5.1.1.original/chapter06/flex.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/flex.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&flex-directory;</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/usr</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter06/gawk.xml LFS-5.1.1/chapter06/gawk.xml
--- LFS-5.1.1.original/chapter06/gawk.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/gawk.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&gawk-directory;</base>
 		</stageinfo>
 
@@ -17,7 +20,9 @@
 			<param>--libexecdir=/usr/lib</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter06/gcc-2953.xml LFS-5.1.1/chapter06/gcc-2953.xml
--- LFS-5.1.1.original/chapter06/gcc-2953.xml	2004-05-18 06:59:04.000000000 +0200
+++ LFS-5.1.1/chapter06/gcc-2953.xml	2004-07-10 11:15:32.000000000 +0200
@@ -14,6 +14,7 @@
 	<stage name="Installing.">
 		<stageinfo>
 			<environment>
+				<variable name="CC_PARALLEL" />
 				<variable name="CFLAGS" />
 				<variable name="CXXFLAGS" />
 			</environment>
@@ -51,6 +52,7 @@
 		</configure>
 
 		<make>
+			<param>-j $CC_PARALLEL</param>
 			<param>bootstrap</param>
 		</make>
 
diff -Naur LFS-5.1.1.original/chapter06/gcc.xml LFS-5.1.1/chapter06/gcc.xml
--- LFS-5.1.1.original/chapter06/gcc.xml	2004-05-18 06:59:04.000000000 +0200
+++ LFS-5.1.1/chapter06/gcc.xml	2004-07-10 11:15:32.000000000 +0200
@@ -27,6 +27,7 @@
 	<stage name="Installing.">
 		<stageinfo>
 			<environment>
+				<variable name="CC_PARALLEL" />
 				<variable name="CFLAGS" />
 				<variable name="CXXFLAGS" />
 			</environment>
@@ -54,7 +55,9 @@
 			<param>--enable-languages=c,c++</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>-k check</param>
diff -Naur LFS-5.1.1.original/chapter06/gettext.xml LFS-5.1.1/chapter06/gettext.xml
--- LFS-5.1.1.original/chapter06/gettext.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/gettext.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&gettext-directory;</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/usr</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter06/glibc.xml LFS-5.1.1/chapter06/glibc.xml
--- LFS-5.1.1.original/chapter06/glibc.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/glibc.xml	2004-07-10 11:15:32.000000000 +0200
@@ -15,6 +15,7 @@
 	<stage name="Installing.">
 		<stageinfo>
 			<environment>
+				<variable name="CC_PARALLEL" />
 				<variable name="CFLAGS" />
 				<variable name="CXXFLAGS" />
 			</environment>
@@ -30,7 +31,9 @@
 			<param>--without-cvs</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>PARALLELMFLAGS="-j $CC_PARALLEL"</param>
+		</make>
 
 		<make>
 			<param>check</param>
diff -Naur LFS-5.1.1.original/chapter06/grep.xml LFS-5.1.1/chapter06/grep.xml
--- LFS-5.1.1.original/chapter06/grep.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/grep.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&grep-directory;</base>
 		</stageinfo>
 
@@ -18,7 +21,9 @@
 			<param>--with-included-regex</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter06/grub.xml LFS-5.1.1/chapter06/grub.xml
--- LFS-5.1.1.original/chapter06/grub.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/grub.xml	2004-07-10 11:15:32.000000000 +0200
@@ -10,6 +10,7 @@
 	<stage name="Installing.">
 		<stageinfo>
 			<environment>
+				<variable name="CC_PARALLEL" />
 				<variable name="CFLAGS" />
 				<variable name="CXXFLAGS" />
 			</environment>
@@ -20,7 +21,9 @@
 			<param>--prefix=/usr</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter06/gzip.xml LFS-5.1.1/chapter06/gzip.xml
--- LFS-5.1.1.original/chapter06/gzip.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/gzip.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&gzip-directory;</base>
 		</stageinfo>
 
@@ -22,7 +25,9 @@
 			<replace>/bin</replace>
 		</search_replace>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter06/inetutils.xml LFS-5.1.1/chapter06/inetutils.xml
--- LFS-5.1.1.original/chapter06/inetutils.xml	2004-05-18 06:59:04.000000000 +0200
+++ LFS-5.1.1/chapter06/inetutils.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&inetutils-directory;</base>
 		</stageinfo>
 
@@ -29,7 +32,9 @@
 			<param>--disable-servers</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter06/kbd.xml LFS-5.1.1/chapter06/kbd.xml
--- LFS-5.1.1.original/chapter06/kbd.xml	2004-05-18 06:59:04.000000000 +0200
+++ LFS-5.1.1/chapter06/kbd.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&kbd-directory;</base>
 		</stageinfo>
 
@@ -20,7 +23,9 @@
 
 		<configure />
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter06/less.xml LFS-5.1.1/chapter06/less.xml
--- LFS-5.1.1.original/chapter06/less.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/less.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&less-directory;</base>
 		</stageinfo>
 
@@ -18,7 +21,9 @@
 			<param>--sysconfdir=/etc</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter06/libtool.xml LFS-5.1.1/chapter06/libtool.xml
--- LFS-5.1.1.original/chapter06/libtool.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/libtool.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&libtool-directory;</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/usr</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter06/m4.xml LFS-5.1.1/chapter06/m4.xml
--- LFS-5.1.1.original/chapter06/m4.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/m4.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&m4-directory;</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/usr</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter06/make.xml LFS-5.1.1/chapter06/make.xml
--- LFS-5.1.1.original/chapter06/make.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/make.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&make-directory;</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/usr</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter06/modutils.xml LFS-5.1.1/chapter06/modutils.xml
--- LFS-5.1.1.original/chapter06/modutils.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/modutils.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,12 +9,17 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&modutils-directory;</base>
 		</stageinfo>
 
 		<configure />
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter06/ncurses.xml LFS-5.1.1/chapter06/ncurses.xml
--- LFS-5.1.1.original/chapter06/ncurses.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/ncurses.xml	2004-07-10 11:15:32.000000000 +0200
@@ -10,6 +10,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&ncurses-directory;</base>
 		</stageinfo>
 
@@ -19,7 +22,13 @@
 			<param>--without-debug</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>sources</param>
+		</make>
+
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter06/patch.xml LFS-5.1.1/chapter06/patch.xml
--- LFS-5.1.1.original/chapter06/patch.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/patch.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+ 			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&patch-directory;</base>
 		</stageinfo>
 
@@ -17,7 +20,9 @@
 			<param>--prefix=/usr</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter06/perl.xml LFS-5.1.1/chapter06/perl.xml
--- LFS-5.1.1.original/chapter06/perl.xml	2004-05-16 20:58:52.000000000 +0200
+++ LFS-5.1.1/chapter06/perl.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&perl-directory;</base>
 		</stageinfo>
 
@@ -17,7 +20,9 @@
 			<param>-Dpager="/bin/less -isR"</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<textdump>
diff -Naur LFS-5.1.1.original/chapter06/procinfo.xml LFS-5.1.1/chapter06/procinfo.xml
--- LFS-5.1.1.original/chapter06/procinfo.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/procinfo.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,10 +9,14 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&procinfo-directory;</base>
 		</stageinfo>
 
 		<make>
+			<param>-j $CC_PARALLEL</param>
 			<param>LDLIBS=-lncurses</param>
 		</make>
 
diff -Naur LFS-5.1.1.original/chapter06/procps.xml LFS-5.1.1/chapter06/procps.xml
--- LFS-5.1.1.original/chapter06/procps.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/procps.xml	2004-07-10 11:15:32.000000000 +0200
@@ -10,10 +10,15 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&procps-directory;</base>
 		</stageinfo>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter06/psmisc.xml LFS-5.1.1/chapter06/psmisc.xml
--- LFS-5.1.1.original/chapter06/psmisc.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/psmisc.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&psmisc-directory;</base>
 		</stageinfo>
 
@@ -17,7 +20,9 @@
 			<param>--exec-prefix=/</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter06/sed.xml LFS-5.1.1/chapter06/sed.xml
--- LFS-5.1.1.original/chapter06/sed.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/sed.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&sed-directory;</base>
 		</stageinfo>
 
@@ -17,7 +20,9 @@
 			<param>--bindir=/bin</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter06/shadow.xml LFS-5.1.1/chapter06/shadow.xml
--- LFS-5.1.1.original/chapter06/shadow.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/shadow.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&shadow-directory;</base>
 		</stageinfo>
 
@@ -28,7 +31,9 @@
 			</content>
 		</textdump>
 		
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter06/sysklogd.xml LFS-5.1.1/chapter06/sysklogd.xml
--- LFS-5.1.1.original/chapter06/sysklogd.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/sysklogd.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,10 +9,15 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&sysklogd-directory;</base>
 		</stageinfo>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 		<make>
 			<param>install</param>
diff -Naur LFS-5.1.1.original/chapter06/sysvinit.xml LFS-5.1.1/chapter06/sysvinit.xml
--- LFS-5.1.1.original/chapter06/sysvinit.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/sysvinit.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&sysvinit-directory;</base>
 		</stageinfo>
 
@@ -19,6 +22,7 @@
 		</search_replace>
 
 		<make>
+			<param>-j $CC_PARALLEL</param>
 			<param>-C src</param>
 		</make>
 
diff -Naur LFS-5.1.1.original/chapter06/tar.xml LFS-5.1.1/chapter06/tar.xml
--- LFS-5.1.1.original/chapter06/tar.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/tar.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&tar-directory;</base>
 		</stageinfo>
 
@@ -18,7 +21,9 @@
 			<param>--libexecdir=/usr/sbin</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter06/texinfo.xml LFS-5.1.1/chapter06/texinfo.xml
--- LFS-5.1.1.original/chapter06/texinfo.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/texinfo.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&texinfo-directory;</base>
 		</stageinfo>
 
@@ -16,7 +19,9 @@
 			<param>--prefix=/usr</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter06/util-linux.xml LFS-5.1.1/chapter06/util-linux.xml
--- LFS-5.1.1.original/chapter06/util-linux.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/util-linux.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&util-linux-directory;</base>
 		</stageinfo>
 
@@ -26,6 +29,7 @@
 		<configure />
 
 		<make>
+			<param>-j $CC_PARALLEL</param>
 			<param>HAVE_KILL=yes</param>
 			<param>HAVE_SLN=yes</param>
 		</make>
diff -Naur LFS-5.1.1.original/chapter06/vim.xml LFS-5.1.1/chapter06/vim.xml
--- LFS-5.1.1.original/chapter06/vim.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/vim.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&vim-directory;</base>
 		</stageinfo>
 
@@ -24,7 +27,9 @@
 			<param>--prefix=/usr</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/chapter06/zlib.xml LFS-5.1.1/chapter06/zlib.xml
--- LFS-5.1.1.original/chapter06/zlib.xml	2004-05-16 01:46:48.000000000 +0200
+++ LFS-5.1.1/chapter06/zlib.xml	2004-07-10 11:15:32.000000000 +0200
@@ -9,6 +9,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&build_dir;/&zlib-directory;</base>
 		</stageinfo>
 
@@ -17,7 +20,9 @@
 			<param>--shared</param>
 		</configure>
 
-		<make />
+		<make>
+			<param>-j $CC_PARALLEL</param>
+		</make>
 
 <!--
 		<make>
diff -Naur LFS-5.1.1.original/config_seperate/general.ent LFS-5.1.1/config_seperate/general.ent
--- LFS-5.1.1.original/config_seperate/general.ent	2004-06-06 02:55:58.000000000 +0200
+++ LFS-5.1.1/config_seperate/general.ent	2004-07-10 11:15:32.000000000 +0200
@@ -134,6 +134,12 @@
 <!-- Directory (inside LFS) where all packages will be compiled. -->
 <!ENTITY build_dir "/tools/build">
 
+<!-- gcc parallel instances
+    see http://www.linuxfromscratch.org/hints/downloads/files/parallelcompiling.txt
+    for more information
+    -->
+<!ENTITY gcc_parallel "1">
+
 <!-- gcc optimizations flags (overridden in packages known
     to behave badly with optimizations 
     see http://www.linuxfromscratch.org/hints/downloads/files/optimization.txt for more info
diff -Naur LFS-5.1.1.original/config_seperate/kernel.xml LFS-5.1.1/config_seperate/kernel.xml
--- LFS-5.1.1.original/config_seperate/kernel.xml	2004-05-16 01:46:49.000000000 +0200
+++ LFS-5.1.1/config_seperate/kernel.xml	2004-07-10 11:15:32.000000000 +0200
@@ -14,6 +14,9 @@
 
 	<stage name="Installing.">
 		<stageinfo>
+			<environment>
+				<variable name="CC_PARALLEL" />
+			</environment>
 			<base>&linux-src-location;/&linux-directory;</base>
 		</stageinfo>
 
@@ -38,11 +41,13 @@
 		</make>
 
 		<make>
+			<param>-j $CC_PARALLEL</param>
 			<param>CC=/opt/gcc-2.95.3/bin/gcc</param>
 			<param>bzImage</param>
 		</make>
 
 		<make>
+			<param>-j $CC_PARALLEL</param>
 			<param>CC=/opt/gcc-2.95.3/bin/gcc</param>
 			<param>modules</param>
 		</make>
